appModules/SearchUI: add 'searchhost' as an alias of Search UI app module in Windows 11

josephsl opened the pull request:
Hi,

Another part of #12585:

### Link to issue number:
Expands #10341 

### Summary of the issue:
Windows Search executable was renamed to searchhost.exe in Windows 11, affecting suggestion count announcement.

### Description of how this pull request fixes the issue:
Add an alias app module named "searchhost.py" to support suggestions count announcement in Windows 11.

### Testing strategy:
Manual testing:

1. In Windows 11 preview, open File Explorer (Windows+E).
2. Press F3 to open search interface.
3. Type something. NVDA should announce suggestion count.

### Known issues with pull request:
None

### Change log entries:
Tweaking #10341 to say:

New features:
In Windows 10 Version 1909 and later (including Windows 11), NVDA will announce suggestion count when performing searches in File Explorer. (#10341, #12628)

### Code Review Checklist:

<!--
This checklist is a reminder of things commonly forgotten in a new PR.
Authors, please do a self-review of this pull-request.
Check items to confirm you have thought about the relevance of the item.
Where items are missing (eg unit / system tests), please explain in the PR.
To check an item `- [ ]` becomes `- [x]`, note spacing.
You can also check the checkboxes after the PR is created.
A detailed explanation of this checklist is available here:
https://github.com/nvaccess/nvda/blob/master/devDocs/githubPullRequestTemplateExplanationAndExamples.md#code-review-checklist
-->

- [x] Pull Request description is up to date.
- [x] Unit tests.
- [x] System (end to end) tests.
- [x] Manual testing.
- [x] User Documentation.
- [x] Change log entry.
- [x] Context sensitive help for GUI changes.
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers

josephsl committed:
SearchUI: add 'searchhost' as an alias of Search UI app module in Windows 11.

Windows Search executable was renamed to searchhost.exe in Windows 11, affecting suggestion count announcement. Therefore add an alias app module to support Windows 11.

seanbudd commented on source/appModules/searchhost.py:
```diff
@@ -0,0 +1,10 @@
+"""Windows Search app module for Windows 11 (build 22000) and later.
+"""
+
+# Flake8/F403: this is an alias for SearchUI app module.
+from .searchui import * # NOQA
```
Assuming that these apps may diverge eventually, I think it is better to explicit. [See the recently started guide for imports, which doesn't perfectly cover this case](https://github.com/nvaccess/nvda/blob/master/devDocs/codingStandards.md#imports).

```suggestion
from .searchui import AppModule, StartMenuSearchField
__all__ = ["AppModule", "StartMenuSearchField"]
```

josephsl committed:
SearchHost: apply newer import guidelines. Re #12628.

Note from Sean Budd (NV Access): use __all__ to specify module attributes and to avoid Flake8 NOQA comment in the process.

Co-authored-by: Sean Budd <seanbudd123@gmail.com>

josephsl committed:
Merge branch 'master' into w11searchhost

seanbudd committed:
update changes
